e2e/docker-run.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Copyright 2017 DigitalOcean
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+PROJECT_DIR=$(dirname ${SCRIPT_DIR})
+
+(
+    cd ${PROJECT_DIR}
+
+    DOCKER_RUN_ARGS=""
+    if [ -f e2e/.env ]; then
+        DOCKER_RUN_ARGS="--env-file e2e/.env"
+    fi
+
+    docker build -t do-ccm-e2e e2e
+    docker run \
+        --rm -it \
+        -v ${PROJECT_DIR}:/app \
+        -w /app \
+        -e DIGITALOCEAN_ACCESS_TOKEN \
+        -e KOPS_REGION \
+        -e S3_ENDPOINT \
+        -e S3_ACCESS_KEY_ID \
+        -e S3_SECRET_ACCESS_KEY \
+        -e KOPS_CLUSTER_NAME \
+        ${DOCKER_RUN_ARGS} \
+        do-ccm-e2e \
+        ./e2e/e2e.sh
+)
```

Comment:
The script seems almost identical to `docker-clean.sh` except for which script is invoked and from where. Should we use a single docker runner at this point?

(For extra convenience, we could have a very small entry point script (similar to [this one](https://success.docker.com/article/use-a-script-to-initialize-stateful-container-data)) to accept parameters like `e2e` and `cleanup` and calls the appropriate scripts.)

Reply:
I like the rule of 3 in these situations (abstract at the 3rd instance), but will DRY this up a bit 😄

Reply:
That's actually a very good rule, so I'm also okay with sticking to the duplication for now. 👍

Reply:
DRYed this up into just a `e2e/docker.sh test|clean` script.